src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs

```diff
@@ -29,14 +45,15 @@ public SearchBoxControl()
         /// <exception cref="NotImplementedException"></exception>
         private void SearchTextBox_OnTextChanged(object sender, TextChangedEventArgs e)
         {
-            var textBox = sender as TextBox;
-            if (textBox == null) return;
-
-            (this.DataContext as PackageManagerSearchViewModel).SearchAndUpdateResults(textBox.Text);
-
-            textBox.Focus();
+            // set delay for this event 500ms
+            Debounce(delayTimer, TimeSpan.FromMilliseconds(500), () =>
```

Comment:
Would you move the `500` out as an internal static int property?

Reply:
I guess I will more this more to private, it is good to go ?

Reply:
Sure, this one resolved.